src/common/Requirements.ts

```diff
@@ -22,21 +23,25 @@ export interface Requirement {
     logger: Logger;
 }
 
-export interface RequirementResult {
+interface RequirementResult {
     duration: number;
     hasPassed: boolean;
     message: string;
     title: string;
 }
 
-export interface SetupTestResult {
+interface RequirementCheckResult {
     hasMetAllRequirements: boolean;
     tests: RequirementResult[];
 }
 
-export interface RequirementList {
+interface RequirementList {
     requirements: Requirement[];
-    executeSetup(): Promise<SetupTestResult>;
+    enabled: boolean;
+}
+
+interface Checks {
```

Comment:
In addition to my previous comments about requirements groups being owned by commands, I'd propose that this interface change a bit, to formalize that relationship by having those commands implement this interface.  So I'm thinking this becomes more like:
```typescript
interface HasRequirements {
    commandRequirements: { [key: string]: RequirementList };
}
```
And then your class definition becomes e.g.:
```typescript
export class Setup extends SfdxCommand implements HasRequirements { }
```
That way each command has the same property name with the same definition, for exposing these requirements.